src/ORM/Filters/SearchFilter.php

```diff
@@ -345,6 +360,17 @@ public function applyAggregate(DataQuery $query, $having)
             ->groupby("\"{$baseTable}\".\"ID\"");
     }
 
+    /**
+     * Check whether this filter matches against a value.
+     */
+    public function matches(mixed $value): bool
+    {
+        // We can't add an abstract method but we want to enforce the method signature for any subclasses
```

Comment:
Why not just use an abstract method?

Reply:
We can't use abstract because that will mean any custom subclasses of `SearchFilter` will need to implement a new method, which makes it a breaking change.